### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by dpinte**
This PR make sure the test pass on Python 3. It adds python 3 testing on travis-ci

### CODE REVIEW

**Comment by dpinte** [comment]
One failing test on Py3 (that I can replicate locally):

```
======================================================================
FAIL: Test fast query for specific cases.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/enthought/encore/encore/storage/tests/locking_filesystem_store_test.py", line 128, in test_fast_query
    assert_items_equal(results, ['file.1', 'file.2'])
AssertionError: Element counts were not equal:
First has 0, Second has 1:  'file.1'
First has 0, Second has 1:  'file.2'
----------------------------------------------------------------------
Ran 549 tests in 47.608s
```

@corranwebster I think this is ready for a first review. There is a lot of uglyness that I would like to have fixed in this PR but it would have made it huge. As a first step, the changes just make things work on Py3 and ensure Py2 still works as expected. For `encore.storage`, the key value store assumptions are that the keys are `str` or `unicode` and the data are `bytes` string.

File: encore/events/event_manager.py [hunk 1]
```diff
@@ -250,7 +258,7 @@ def get_listeners(self, event):
                 listener = linfo[-1]
                 id = self.get_id(listener())
                 if id in l_filter:
-                    for key, value in l_filter[id].iteritems():
+                    for key, value in six.iteritems(l_filter[id]):
```
Comment (corranwebster) [inline encore/events/event_manager.py]: I'd be willing to live with the simpler `.items()` here (even though less efficient on 2.7) unless we _know_ that creating an intermediate list is a bottleneck for a real application.

File: encore/events/tests/test_event_manager.py [hunk 2]
```diff
@@ -396,7 +398,7 @@ def callback(evt):
         evt = MyEvt()
         self.evt_mgr.connect(BaseEvent, callback)
         self.evt_mgr.emit(evt)
-        self.assertEqual(call_seq, range(3))
+        self.assertEqual(call_seq, list(six.moves.range(3)))
```
Comment (corranwebster) [inline encore/events/tests/test_event_manager.py]: Why not just `list(range(3))`?  Slight inefficiency on 2.7, but this is a test.

**Comment by dpinte** [comment]
@corranwebster all green now, can you review again? 
@jordanweaver this is the first piece missing for `enclosure` in py3

**Comment by corranwebster** [comment]
LGTM other than the comments above.

**Comment by dpinte** [comment]
@corranwebster comments addressed, CI all green, merging!

**Pull request merged**